server/src/routes/collectiveStatement.ts

```diff
@@ -0,0 +1,597 @@
+        group_consensus: groupConsensus[comment.comment_id] || 0
+      }));
+      
+      // Filter out comments with less than 20 votes
+      filteredComments = filteredComments.filter(c => c.total_votes >= 20);
+      
+      // Sort by group consensus (descending)
+      filteredComments.sort((a, b) => b.group_consensus - a.group_consensus);
+      
+      // Take top quartile
+      const quartileSize = Math.ceil(filteredComments.length / 4);
+      filteredComments = filteredComments.slice(0, quartileSize);
+      
+      logger.info(`Filtered from ${topicComments.length} to ${filteredComments.length} comments (min 20 votes, top quartile by consensus)`);
+    }
+    
+    // Generate the collective statement
+    const result = await generateCollectiveStatement(
+      zid,
+      topic_key,
+      topic_name,
+      filteredComments
+    );
+
+    // Store in DynamoDB
+    const item = {
+      zid_topic_jobid: statementKey,
+      zid: zid.toString(),
+      topic_key: topic_key,
+      topic_name: topic_name,
+      statement_data: JSON.stringify(result.statementData),
+      comments_data: JSON.stringify(result.commentsData),
+      created_at: new Date().toISOString(),
+      model: "claude-opus-4-20250514",
+    };
+
+    await docClient.send(new PutCommand({
+      TableName: "Delphi_CollectiveStatement",
+      Item: item,
+    }));
+
+    return res.json({
+      status: "success",
+      statementData: result.statementData,
+      commentsData: result.commentsData,
+      id: statementKey,
+    });
+
+  } catch (err: any) {
+    logger.error(`Error in handle_POST_collectiveStatement: ${err.message}`);
+    logger.error(`Error stack: ${err.stack}`);
+    
+    return res.status(500).json({
+      status: "error",
+      message: "Error generating collective statement",
+      error: err.message,
+    });
+  }
+}
+
+
+/**
+ * Handler for GET /api/v3/collectiveStatement
+ * Can get a single statement by ID or all statements for a report
+ */
+export async function handle_GET_collectiveStatement(req: Request, res: Response) {
+  const { statement_id, report_id } = req.query;
+  
+  // If report_id is provided, get all statements for that conversation
+  if (report_id) {
+    try {
+      const zid = await getZidFromReport(report_id as string);
+      if (!zid) {
+        return res.status(404).json({
+          status: "error",
+          message: "Could not find conversation for report_id",
+        });
+      }
+
+      // Query all items where zid matches
+      const params = {
+        TableName: "Delphi_CollectiveStatement",
+        FilterExpression: "zid = :zid",
+        ExpressionAttributeValues: {
+          ":zid": zid.toString()
+        }
+      };
+
+      const statements: any[] = [];
+      let lastEvaluatedKey;
+
+      do {
+        const command: any = {
+          ...params,
+          ExclusiveStartKey: lastEvaluatedKey
+        };
+        
+        const data = await docClient.send(new ScanCommand(command));
```

Comment:
ok regarding the scan here - we should be able to add a GSI for zid which will prevent the need to scan